### PR TITLE
Beta document changes for Jane.

### PR DESCRIPTION
**Pull request by rykal-nate**
### Checklist:

- [ ] Read the [Code of Conduct](https://github.com/appsody/website/blob/master/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/appsody/website/blob/master/CONTRIBUTING.md).

- [ ] Followed the [commit message guidelines](https://github.com/appsody/website/blob/master/CONTRIBUTING.md#commit-message-guidelines).

- [ ] Stack adheres to [Appsody stack structure](https://github.com/appsody/website/blob/master/content/docs/stacks/stacks-overview.md#stack-structure).

### Modifying an existing stack:

- [ ] Updated the stack version in `stack.yaml`

### Contributing a new stack:

- Describe how application dependencies are managed:

- Explain how Appsody file watcher is utilized:

- Describe other Appsody environment variables defined by the stack image:

- Describe any limitations and known issues:

### CODE REVIEW

File: incubator/blockchain-client-openliberty/README.md [hunk 1]
```diff
@@ -1,95 +1,49 @@
...
 
 The Open Liberty stack uses a parent Maven project object model (POM) to manage dependency versions and provide required capabilities and plugins.
 
-This stack is based on OpenJDK with container-optimizations in OpenJ9 and `Open Liberty v19.0.0.12`. It provides live reloading during development by utilizing the "dev mode" capability in the liberty-maven-plugin.  To see dev mode in action (though not in Appsody) check out this [shorter demo](https://openliberty.io/blog/2019/10/22/liberty-dev-mode.html) and this  [a bit longer demo](https://blog.sebastian-daschner.com/entries/openliberty-plugin-dev-mode).
+This stack is based on OpenJDK with container-optimizations in OpenJ9 and `Open Liberty v20.0.0.3`. It provides live reloading during development by utilizing the "dev mode" capability in the liberty-maven-plugin.  To see dev mode in action (though not in Appsody) check out this [shorter demo](https://openliberty.io/blog/2019/10/22/liberty-dev-mode.html) and this  [a bit longer demo](https://blog.sebastian-daschner.com/entries/openliberty-plugin-dev-mode).
```
Comment (chilanti) [inline incubator/blockchain-client-openliberty/README.md:7]: I would suggest avoiding direct references to the levels of Liberty - just say, it's an extension of the "openliberty" Kabanero stack (we'll have to fix this later).

File: incubator/blockchain-client-openliberty/README.md [hunk 2]
```diff
@@ -1,95 +1,49 @@
...
 
 **Note:** Maven is provided by the Appsody stack container, allowing you to build, test, and debug your Java application without installing Maven locally. However, we recommend installing Maven locally for the best IDE experience.
 
+## Prerequisites
+
+This stack requires the [Open Liberty Operator](https://github.com/OpenLiberty/open-liberty-operator) to be installed in the cluster prior to deploying the stack.
```
Comment (chilanti) [inline incubator/blockchain-client-openliberty/README.md:13]: The Operator is only needed when you deploy it on an Openshift cluster - let's delay that discussion to the deployment section.

File: incubator/blockchain-client-openliberty/README.md [hunk 3]
```diff
@@ -1,95 +1,49 @@
...
-    1. Initialize the stack, e.g. `appsody init dev.local/blockchain-client-openliberty` 
-1. Run the stack
-    1. Find the network name of the local fabric: `docker network ls`, e.g. `1OrgLocalFabric_network`
-    1. Find the location of your `blockchain.env` file, e.g. `/Users/tnixa/blockchain/blockchain.env`
-    1. Run the stack, e.g. `appsody run --network <network name> --docker-options "--env-file=<path to env file>"`
-    
-
-
-# Open Liberty Stack
+# Blockchain Open Liberty Stack
```
Comment (chilanti) [inline incubator/blockchain-client-openliberty/README.md:1]: We need to agree on a name/title. @jorgedr94 ?

File: incubator/blockchain-client-openliberty/README.md [hunk 4]
```diff
@@ -180,6 +132,34 @@ The metrics endpoint is secured with a userid and password enabled through the c
 
 In order to lock down the production image built via `appsody build` this file is deleted during the Docker build of your application production image.  (The same file would be deleted if you happened to create your own file at this location as well).
 
+## Stack development
```
Comment (chilanti) [inline incubator/blockchain-client-openliberty/README.md:135]: Is this section meant to be the "Champ" section about extending the stack?

File: incubator/blockchain-client-openliberty/templates/default/README.md [hunk 5]
```diff
@@ -0,0 +1,96 @@
+# Default template for Blockchain with Open Liberty
+
+## Introduction
+
+The intent of the default template in the Blockchain Open Liberty stack is to provide a working example that easily integrates with the default Smart Contract provided by The IBM® Blockchain Platform Visual Studio (VS) Code extension.(<https://cloud.ibm.com/docs/services/blockchain/howto?topic=blockchain-develop-vscode)>
+
+That default Smart Contract is called MyAsset and following a minimal set of instructions detailed in the link above you can have a Fabric Network running the MyAsset Smart Contract.  This template provides an means to easily bind a Java Openlibery Service to the MyAsset Smart Contract. For learning purposes your new project is able to run, connect and execute transactions without any code changes. Not only can you bind to a Local Fabric Instance created by VSCode, you can in fact interact with any Blockchain Fabric Network instance.(<https://www.ibm.com/blockchain/platform)>
```
Comment (chilanti) [inline incubator/blockchain-client-openliberty/templates/default/README.md]: `This template provides an means...` -> `This template provides a means...`

Comment (chilanti) [reply incubator/blockchain-client-openliberty/templates/default/README.md]: `easily bind a Java Openlibery Service` -> `expose a JAX-RS REST service` (perhaps?)

File: incubator/blockchain-client-openliberty/templates/default/README.md [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+
+The template provides a well structured web application with common componently to get you starting with all things common to transacting on a fabric network.  This starting point is for you to refactor, extend and customize to meet your own project requirements. There are a few component layers that you will want to refactor for your projects needs and a few to only configure. The API Layer, Identity Mapping Layer and finally the Transaction Management Layer. In the following sections we will define layer.
```
Comment (chilanti) [inline incubator/blockchain-client-openliberty/templates/default/README.md]: `with common componently`-> components or componentry?

File: incubator/blockchain-client-openliberty/templates/default/README.md [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+
+## Required Envirnment Variables
```
Comment (chilanti) [inline incubator/blockchain-client-openliberty/templates/default/README.md:41]: Typo here.

Comment (chilanti) [reply incubator/blockchain-client-openliberty/templates/default/README.md:41]: I would move this section and the next after the explanation of the layers.

File: incubator/blockchain-client-openliberty/templates/default/README.md [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+
+Wallet(s) are a concept in the fabric SDK that contain identities for transacting on your fabric network. To remain consistent with the terminology we have created the a WalletManager class that contains code for easily onboarding fabric identities into the service.  The WalletManager is intensionally extendable so other wallet implementations can easily be made use of. The purpose of this component is house all the fabric identifies your service will want to transact with. The approach is entirely configuration driven, there are 2 modes of operation out of the box.
+
+`FABRIC_WALLET_PROFILE={"type":"IN_MEMORY"}`
+
+`FABRIC_WALLET_PROFILE={"type":"FILE_SYSTEM","options":{"path":"/project/user-app/src/main/java/application/Org1"}}`
+
+Creating your own Wallet backend is as simple as creating Wallet Implementation, adding configuration and adding a configuration type so the wallet manager can load the new implementation when configured.
+
+## API and Data Model
```
Comment (chilanti) [inline incubator/blockchain-client-openliberty/templates/default/README.md]: This should be the first section, before wallets.

File: incubator/blockchain-client-openliberty/templates/default/README.md [hunk 9]
```diff
@@ -0,0 +1,96 @@
...
+
+The API Layer is the REST API definition layer for the micro service. This layer defines the set of input and output for each REST API endpoints the service exposes. In the template this component is simply AssetService.java. Looking at the class you will notice a set of open api annotations for each function. It is a best practice to use the Open API specification to document each endpoint, this enables an easy way for developers to test and also provides self service documentation to client consumers of the service.
+
+The Data Model layer are the Java Object that represent the set of inputs and output you wish to model. In the sample this component is simply Asset.java.
+
+## Identity Mapping
+
+Every service should be secure and provide a mechanism for authentication. The strategies you choose are beyond the scope of what the application template can provide, however we thought it important there be a minimal set of components that address the issue of mapping a service user to a user of an organization of a fabric network that the service user is acting on behalf. Here is the explanation of how the provider identity parts are meant to be use …
```
Comment (chilanti) [inline incubator/blockchain-client-openliberty/templates/default/README.md]: Somewhere, I think we should add something like: "The template that is provided with the stack makes no assumptions on the specific authentication mechanism used by the client. It assumes that the client invoking the microservice that is based on the template is a trusted client, and that it will have performed some form of authentication. The trusted client will then forward an authentication token - such as JSON Web Token or a SAML assertion - to the microservice. It is the developer's responsibility to extract the appropriate claims from that token and map those claims to a Fabric identity. The template provided with the stack only provides the building blocks to implement this logic - but no specific implementation."

File: incubator/blockchain-client-openliberty/templates/default/README.md [hunk 10]
```diff
@@ -0,0 +1,96 @@
...
+[Container] [INFO] BUILD FAILURE
+[Container] [INFO] ------------------------------------------------------------------------
+[Container] [INFO] Total time: 59.154 s
+[Container] [INFO] Finished at: 2020-04-22T20:28:52Z

+[Container] [INFO] ------------------------------------------------------------------------
+[Container] [ERROR] Failed to execute goal io.openliberty.tools:liberty-maven-plugin:3.2:dev (default-cli) on project starter-app: Could not parse the host name from the log message: {"type":"liberty_message","host":"b10c3a9ac6d0","ibm_userDir":"/opt/ol/wlp/usr/","ibm_serverName":"defaultServer","message":"CWWKT0016I: Web application available (default_host): http://b10c3a9ac6d0:9080/metrics/","ibm_threadId":"00000034","ibm_datetime":"2020-04-22T20:28:22.369+0000","ibm_messageId":"CWWKT0016I","module":"com.ibm.ws.http.internal.VirtualHostImpl","loglevel":"AUDIT","ibm_sequence":"1587587302369_000000000001F"} -> [Help 1]
+```
+
+## Deploy
```
Comment (chilanti) [inline incubator/blockchain-client-openliberty/templates/default/README.md:96]: This is where we want to talk about the pre-req's.

**Review by chilanti — CHANGES_REQUESTED** [review]
@rykal-nate - really good first pass - let me know if you want to meet and discuss.

File: incubator/blockchain-client-openliberty/README.md (hunk 1 above)
Comment (rykal-nate) [reply incubator/blockchain-client-openliberty/README.md:7]: This is exactly what is in the parent readme, so I tried to keep everything the base liberty users would expect to see.

File: incubator/blockchain-client-openliberty/README.md (hunk 2 above)
Comment (rykal-nate) [reply incubator/blockchain-client-openliberty/README.md:13]: This is exactly what is in the parent readme, so I tried to keep everything the base liberty users would expect to see.

File: incubator/blockchain-client-openliberty/README.md (hunk 4 above)
Comment (rykal-nate) [reply incubator/blockchain-client-openliberty/README.md:135]: Yes - just a placeholder.

File: incubator/blockchain-client-openliberty/templates/default/README.md (hunk 5 above)
Comment (rykal-nate) [reply incubator/blockchain-client-openliberty/templates/default/README.md]: Thanks will make those changes.

File: incubator/blockchain-client-openliberty/templates/default/README.md (hunk 6 above)
Comment (rykal-nate) [reply incubator/blockchain-client-openliberty/templates/default/README.md]: Agree, will come up with something better.

File: incubator/blockchain-client-openliberty/templates/default/README.md (hunk 7 above)
Comment (rykal-nate) [reply incubator/blockchain-client-openliberty/templates/default/README.md:41]: Sounds good!

File: incubator/blockchain-client-openliberty/templates/default/README.md (hunk 8 above)
Comment (rykal-nate) [reply incubator/blockchain-client-openliberty/templates/default/README.md]: Sounds good!

File: incubator/blockchain-client-openliberty/templates/default/README.md (hunk 9 above)
Comment (rykal-nate) [reply incubator/blockchain-client-openliberty/templates/default/README.md]: Thanks for the writeup

**Review by chilanti — APPROVED** [review]
LGTM

File: incubator/blockchain-client-openliberty/templates/default/README.md [hunk 11]
```diff
@@ -0,0 +1,96 @@
+# Default template for Blockchain with Open Liberty
+
+## Introduction
+
+The intent of the default template in the Blockchain Open Liberty stack is to provide a working example that easily integrates with the default Smart Contract provided by The IBM® Blockchain Platform Visual Studio (VS) Code extension.(<https://cloud.ibm.com/docs/services/blockchain/howto?topic=blockchain-develop-vscode)>
```
Comment (jorgedr94) [inline incubator/blockchain-client-openliberty/templates/default/README.md:5]: ... is to provide the basis for an REST API micro-service that easily integrates ...

File: incubator/blockchain-client-openliberty/templates/default/README.md [hunk 12]
```diff
@@ -0,0 +1,96 @@
...
+
+That default Smart Contract is called MyAsset and following a minimal set of instructions detailed in the link above you can have a Fabric Network running the MyAsset Smart Contract.  This template provides a means to easily expose a JAX-RS REST service to the MyAsset Smart Contract. For learning purposes your new project is able to run, connect and execute transactions without any code changes. Not only can you bind to a Local Fabric Instance created by VSCode, you can in fact interact with any Blockchain Fabric Network instance.(<https://www.ibm.com/blockchain/platform)>
+
+The template provides a well structured web application with common components to get you starting with all things common to transacting on a fabric network.  This starting point is for you to refactor, extend and customize to meet your own project requirements. There are a few component layers that you will want to refactor for your projects needs and a few to only configure. The API Layer, Identity Mapping Layer and finally the Transaction Management Layer. In the following sections we will define layer.
```
Comment (jorgedr94) [inline incubator/blockchain-client-openliberty/templates/default/README.md:9]: Fabric should always be capitalized,  in fact we  should use Hyperledger Fabric instead

Comment (jorgedr94) [reply incubator/blockchain-client-openliberty/templates/default/README.md:9]: ???In the following sections we will define layer???

File: incubator/blockchain-client-openliberty/templates/default/README.md [hunk 13]
```diff
@@ -0,0 +1,96 @@
...
+
+## API and Data Model
+
+The API Layer is the REST API definition layer for the micro service. This layer defines the set of input and output for each REST API endpoints the service exposes. In the template this component is simply AssetService.java. Looking at the class you will notice a set of open api annotations for each function. It is a best practice to use the Open API specification to document each endpoint, this enables an easy way for developers to test and also provides self service documentation to client consumers of the service.
```
Comment (jorgedr94) [inline incubator/blockchain-client-openliberty/templates/default/README.md:13]: Formatting: `AssetService.java`

File: incubator/blockchain-client-openliberty/templates/default/README.md [hunk 14]
```diff
@@ -0,0 +1,96 @@
...
+
+The Data Model layer are the Java Object that represent the set of inputs and output you wish to model. In the sample this component is simply Asset.java.
```
Comment (jorgedr94) [inline incubator/blockchain-client-openliberty/templates/default/README.md:15]: Formatting: `Asset.java`

File: incubator/blockchain-client-openliberty/templates/default/README.md [hunk 15]
```diff
@@ -0,0 +1,96 @@
...
+In a more sophisticated implementation, the mapper would call out to a rule engine, or to a database to perform the mapping.
+1. The identity is then set in an HTTP Header named X-FABRIC-IDENTITY.
+1. The request now gets routed to the API resource.
+1. The resource logic extracts the identity from the X-FABRIC-IDENTITY header and uses it to call getContract(identity) on the ConnectionManager.
+
+## Required Envirnment Variables
+
+Here is a complete list of the supported environment variables as well as an example configuration.
+
+***FABRIC_CONNECTION_PROFILE*** : Blockchain standard document that determines how a gateway connects to a fabric network.  <https://hyperledger-fabric.readthedocs.io/en/release-2.0/developapps/connectionprofile.html> This document should be "compressed/minified" so that there are no spaces or line breaks.
```
Comment (jorgedr94) [inline incubator/blockchain-client-openliberty/templates/default/README.md:45]: ....: Standard JSON document used to described components in a Hyperledger Fabric network and used by the micro-service to configure a gateway that connects to the blockchain network.

File: incubator/blockchain-client-openliberty/templates/default/README.md [hunk 16]
```diff
@@ -0,0 +1,96 @@
...
+
+***FABRIC_WALLET_PROFILE*** : Setting to specify how the service finds and loads the identies. There are two supported types out of the box. In memory or filesystem. This document should be "compressed/minified" so that there are no spaces or line breaks.
+
+`{"type":"IN_MEMORY"}`
+
+`{"type":"FILE_SYSTEM","options":{"path":"/project/user-app/src/main/java/application/Org1"}}`
+
+option.path : is that path to the file system wallet structure in compliance with the SDK wallet structure. This file MUST be included in the project and is for ***DEVELOPMENT PURPOSES ONLY***.
```
Comment (jorgedr94) [inline incubator/blockchain-client-openliberty/templates/default/README.md:53]: Formatting: `option.path`

Comment (jorgedr94) [reply incubator/blockchain-client-openliberty/templates/default/README.md:53]: I am not sure there is a need to say this is for development purposes only.  That all depends on the level of risk that the deployer wants to take and the level of security where the systems are deployed.   I would remove that part.

File: incubator/blockchain-client-openliberty/templates/default/README.md [hunk 17]
```diff
@@ -0,0 +1,96 @@
...
+
+***FABRIC_WALLET_CREDENTIALS*** : A JSON array of fabric identies used when using IN_MEMORY wallet type. (See sample below). This document should be "compressed/minified" so that there are no spaces or line breaks. Additionally, the cert and private_key attribute values ***MUST be base64 encoded.***
+
+***FABRIC_CHANNEL*** : Fabric network channel
+***FABRIC_CONTRACT*** : Installed and Instantiated Smart Contract on the FABRIC_CHANNEL
```
Comment (jorgedr94) [inline incubator/blockchain-client-openliberty/templates/default/README.md:58]: instantiated smart contract should be lower case.    Technically a smart contract is not installed on a channel,  smart contracts are instantiated in a channel so I would rephrase that to avoid the inaccuracy in the sentence.

File: incubator/blockchain-client-openliberty/templates/default/README.md [hunk 18]
```diff
@@ -0,0 +1,96 @@
...
+***FABRIC_DEFAULT_IDENTITY*** : The default identity to use when transacting with the fabric network.
```
Comment (jorgedr94) [inline incubator/blockchain-client-openliberty/templates/default/README.md:59]: There is a correlation here between the content of FABRIC_WALLET_CREDENTIALS and also the wallet we are using.  We should be clear about this. Basically in a more complex environment where API callers are mapped to Fabric identities there is really no need for this default identity.   In that scenario an authenticated API caller will map to a fabric identity and the IdentityMapper component would do that for you.  In the case of this basic example since the API is not being authenticated we use this FABRIC_DEFAULT_IDENTITY to tell us what should be the name of the identity we should use by default.   This value should map to the name field of one of the identities injected through FABRIC_WALLET_CREDENTIALS or the name field of one of the identities found in the filesystem when using the Filesystem Wallet.
